### PR TITLE
Add integrity hash for nightly builds

### LINKED ISSUE #14: Add checksums for downloads
We should publish hash-based checksums to allow verifying downloads.

### PR DESCRIPTION
**Pull request by camelid**
Compute a SHA-256 integrity hash for nightly builds on the download.servo.org page.

---

- [ ] `./mach build -d` does not report any errors (it does, but unrelated to the change)
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix servo/download.servo.org#14 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they're in the infrastructure

### CODE REVIEW

**Comment by camelid** [comment]
Feel free to suggest changes, I'm still unsure about how the infrastructure works!

File: python/servo/package_commands.py [hunk 1]
```diff
@@ -630,12 +631,33 @@ def upload_to_s3(platform, package, timestamp):
...
+            sha256_digest = hashlib.sha256()
+            with open(package, 'rb') as package_file:
+                while True:
+                    data = package_file.read(SHA_BUF_SIZE)
+                    if not data:
+                        break
+                    sha256_digest.update(data)
+            package_hash = sha256_digest.hexdigest()
+            package_hash_fileobj = io.BytesIO(package_hash)
+            package_hash_upload_key = '{}/{}.sha256'.format(nightly_dir, filename)
```
Comment (jdm) [inline python/servo/package_commands.py:645]: I don't think it's important to preserve hashes for each nightly build. Let's just upload the latest hash.

**Review by jdm — CHANGES_REQUESTED** [review]

File: python/servo/package_commands.py (hunk 1 above)
Comment (camelid) [reply python/servo/package_commands.py:645]: Okay, sure!

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by camelid** [comment]
Thanks! :)

**Comment by camelid** [comment]
@jdm For some reason, one of the tests timed-out. I tried to tell bors to retry it, but apparently I need privileges :). Could you tell bors to retry it?

**Comment by jdm** [comment]
@bors-servo retry
* #24622

**Comment by CYBAI** [comment]
@bors-servo retry

- #23290

**Comment by camelid** [comment]
Thanks all!

**Pull request merged**